cine-flex/src/components/HOC/withAd.jsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from "react";
+
+const withAd = (
+  WrappedComponent,
+  timeLeftForAdStart,
+  timeLeftForAdStop,
+) => {
+
+  return (props) => {
+    const [isAdVisible, setIsAdVisible] = useState(false);
+    const [selectedAd, setSelectedAd] = useState(null);
+    const [secsLeftForAdStart, setSecsLeftForAdStart] = useState(timeLeftForAdStart);
+    const [secsLeftForAdStop, setSecsLeftForAdStop] = useState(timeLeftForAdStop);
+    const [isVideoPlaying, setIsVideoPlaying] = useState(false);
+    const [isAdEnded, setIsAdEnded] = useState(false);
+
+    useEffect(() => {
+        if(isVideoPlaying && secsLeftForAdStart >= 0) {
+            decrementTimeLeftForAdStart();
+        }
+    },[secsLeftForAdStart, isVideoPlaying]);
+
+    const decrementTimeLeftForAdStop = () => {
+        setTimeout(() => {
+            clearAdStopTimer();
+            stopAd();
+        }, timeLeftForAdStop*1000);
+
+        const adStopTimer = setInterval(() => {
+            if(secsLeftForAdStop >= 0) {
+                setSecsLeftForAdStop((prevState) => prevState - 1);
+            }
+        }, 1000);
+
+        const clearAdStopTimer = () => {
+            clearInterval(adStopTimer);
+        }
+    };
+
+    const showAd = () => {
+      setSelectedAd(Math.ceil(Math.random() * 2));
+      setIsAdVisible(true);
+      setIsVideoPlaying(false);
+      decrementTimeLeftForAdStop();
+    };
+
+    const decrementTimeLeftForAdStart = () => {
+      setIsVideoPlaying(true);
+      setTimeout(() => {
+        if (secsLeftForAdStart > 0 && isVideoPlaying) {
+          setSecsLeftForAdStart((prevState) => prevState - 1);
+        } else if (secsLeftForAdStart === 0) {
+          showAd();
+        } 
+      }, 1000);
+    };
+
+    const stopAd = () => {
+      setIsAdVisible(false);
```

Comment:
kindly explain when advisible false means 
video should be playing & ad is ended
do we need separate states for these?

Reply:
I have isAdEnded and isAdVisible states to make rendering 'Ad in ... secs' in the UI